runtime/lib/backend/lightning/lightning_dynamic/LightningSimulator.cpp

```diff
@@ -104,6 +104,19 @@ void LightningSimulator::PrintState()
     cout << state[idx] << "]" << endl;
 }
 
+void LightningSimulator::SetState(DataView<std::complex<double>, 1> &data)
+{
+    std::vector<std::complex<double>> data_vector(data.begin(), data.end());
+    std::vector<std::size_t> count(data_vector.size());
+    std::iota(std::begin(count), std::end(count), 0);
```

Comment:
Okay I guess we addressed this offline, the reason for the confusion here was that the frontend was handling the conversion from `wires` + `elements` into a single element array with respect to all wires. Since this goes against how most other operations in the interface are designed the conversion (which is specific to lightning's implementation) should be handled by the device plugin itself.